include/eeprom/core/accessor.hpp

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include <array>
+#include <cassert>
+#include <vector>
+
+#include "addresses.hpp"
+#include "common/core/bit_utils.hpp"
+#include "task.hpp"
+#include "types.hpp"
+
+namespace eeprom {
+namespace accessor {
+
+class AccessorBuffer {
+  public:
+    AccessorBuffer() = default;
+    template <typename B_ITER, typename L_ITER>
+    requires bit_utils::ByteIterator<B_ITER> &&
```

Comment:
kind of a nit but we should require https://en.cppreference.com/w/cpp/iterator/contiguous_iterator if we're going to alias this with a pair of raw pointers